cartoframes/viz/map.py

```diff
@@ -190,8 +191,47 @@ def publish(self, name, maps_api_key='default_public', context=None, password=No
             raise CartoException('The map layers are not synchronized with CARTO. '
                                  'Please, use the `sync_data` method before publishing the map')
 
+        if maps_api_key == 'default_public' and not self._publisher.is_public():
+            raise CartoException('The datasets used in your map are not public. '
+                                 'You need add new Regular API key with permissions to Maps API and the datasets. '
+                                 'You can do it from your CARTO dashboard or using the Auth API. You can get more '
+                                 'info at https://carto.com/developers/auth-api/guides/types-of-API-Keys/')
+
+
         self._publisher.set_context(context)
+        html = self._get_publication_html(name, maps_api_key)
+        self._kuviz = self._publisher.publish(html, name, password)
+        self._kuviz_message()
+
+    def sync_data(self, table_name, context=None):
+        if not self._publisher.is_sync():
+            self._publisher.sync_layers(table_name, context)
+
+    def delete_publication(self):
+        if self._kuviz:
+            self._kuviz.delete()
+            self._kuviz = None
+            print("Publication deleted")
```

Comment:
:+1:

Reply:
About your main comment:

> sync_data is a great method, but do you think users will use that or the more explicit methods in the Dataset class?

No, I don't. It is easier to call `map.sync_data()` than make loops over deep objects of the map class.

> Legends didn't work

It could be a bug? I will ask for some help if I need it

Reply:
I have checked some legends examples and works fine or at least in the same way as the map in the notebook https://cartovl.carto.com/kuviz/88904de8-6e3b-4f6f-af7b-692aabeebb0a

Reply:
Weird, not appearing correctly in my browser or on my mobile. Map displays normally.
<img width="757" alt="Screen Shot 2019-06-11 at 8 22 07 AM" src="https://user-images.githubusercontent.com/1041056/59271567-17f1a580-8c22-11e9-97d6-508f843a9d6c.png">

Reply:
I am seeing the same error in the notebook map, so it is probably a problem in the VL side.

Reply:
There is an issue with jinja2. So please, update this branch with develop and run `pip install -r requirements.txt`

Reply:
We need to focus on this PR please :)

Reply:
I just pointed out the problem and solution of what you are talking about :wink:

Reply:
Sure, thanks!

Reply:
@oleurud, what about [the comment](https://github.com/CartoDB/cartoframes/pull/745/files#r292146249) about adding the name and id to the message about a publication being deleted?

Reply:
OMG, I forgot it!

Reply:
Done (sorry for move you here again)